applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -0,0 +1,900 @@
+            array_1d<double,3>& r_current_residual_moment          = i->FastGetSolutionStepValue(MOMENT_RESIDUAL);
+            array_1d<double,3>& r_current_angular_velocity         = i->FastGetSolutionStepValue(ANGULAR_VELOCITY);
+            //array_1d<double,3>& current_rotation                 = i->FastGetSolutionStepValue(ROTATION);
+            array_1d<double,3>& r_middle_angular_velocity          = i->GetValue(MIDDLE_ANGULAR_VELOCITY);
+            array_1d<double,3>& r_current_angular_acceleration     = i->FastGetSolutionStepValue(ANGULAR_ACCELERATION);
+
+            
+            for (int kk = 0; kk<3; ++kk)
+            {         
+              if (nodal_inertia[kk] > numerical_limit)  r_current_angular_acceleration[kk] = r_current_residual_moment[kk] / nodal_inertia[kk];
+              else r_current_angular_acceleration[kk] = 0.00;
+            }
+
+            DoF = 2;
+            bool fix_rotation[3] = {false, false, false};
+            fix_rotation[0] = (i->pGetDof(ROTATION_X))->IsFixed();
+            fix_rotation[1] = (i->pGetDof(ROTATION_Y))->IsFixed();   
+            
+            
+            if (i->HasDofFor(ROTATION_Z))
+            {
+              DoF = 3;
+              fix_rotation[1] = (i->pGetDof(ROTATION_Z))->IsFixed();  
+            }
+
+            for (int j = 0; j < DoF; j++)
+            {
+              if (fix_rotation[j])
+              {
+                r_current_angular_acceleration[j] = 0.00;
+                r_middle_angular_velocity[j] = 0.00;
+              }
+               
+              r_middle_angular_velocity[j]   = 0.00 + (mTime.Middle - mTime.Previous) * r_current_angular_acceleration[j] ; 
+              r_current_angular_velocity[j]  = r_middle_angular_velocity[j] + (mTime.Previous - mTime.PreviousMiddle) * r_current_angular_acceleration[j];
+              //current_rotation[j]          = 0.00;
+            }//for DoF
+          }// Rot DoF
+
+
+
+
+        }//for node
+
+      }//parallel
+
+      mTime.Previous = mTime.Current;
+      mTime.PreviousMiddle = mTime.Middle;
+      KRATOS_CATCH("")
+    }
+
+
+  //***************************************************************************
+  //***************************************************************************
+  void Calculate_RHS_Contribution(Element::Pointer pCurrentElement,
+          LocalSystemVectorType& RHS_Contribution,
+          Element::EquationIdVectorType& EquationId,
+          ProcessInfo& rCurrentProcessInfo) 
+  {
+
+    KRATOS_TRY
+
+    //basic operations for the element considered
+    Matrix dummy_lhs;
+    pCurrentElement -> CalculateLocalSystem(dummy_lhs,RHS_Contribution,rCurrentProcessInfo);
+
+    //add explicit contribution of the Element Residual (RHS) to nodal Force Residual (nodal RHS)
+    (pCurrentElement) -> AddExplicitContribution(RHS_Contribution, RESIDUAL_VECTOR, FORCE_RESIDUAL, rCurrentProcessInfo);
+    if (pCurrentElement->GetGeometry()[0].HasDofFor(ROTATION_X))
```

Comment:
find another way to make this check, without requiring the dof if you can

Reply:
fixed